contracts/FindUserStatus.cdc

```diff
@@ -0,0 +1,296 @@
+		// Total fee needed for repayment
+        pub var repaymentAmount: UFix64
+        pub var renter: Address? 
+		pub var renterName: String? 
+
+		pub let tags: {String : String} 
+		pub let scalars: {String : UFix64}
+		pub let extra: {String : AnyStruct}
+
+		init(flowtyStorefrontID: UInt64, rented: Bool, nftType: String, nftID: UInt64, amount: UFix64, deposit: UFix64, term: UFix64, paymentVaultType: String, reenableOnReturn:Bool, paymentCuts: [StoreFrontCut], listedTime: UFix64, royaltyRate: UFix64, expiresAfter: UFix64, repaymentAmount: UFix64, renter: Address?) {
+			self.listingId = flowtyStorefrontID
+			self.rented = rented
+			self.nftIdentifier = nftType
+			self.nftId = nftID
+			self.deposit = deposit
+			self.amount = amount
+			self.term = term
+			self.paymentVaultType = paymentVaultType
+			self.reenableOnReturn = reenableOnReturn
+			self.paymentCuts = paymentCuts
+			self.listedTime = listedTime
+			self.royaltyRate = royaltyRate
+			self.listingValidUntil = expiresAfter + listedTime
+			self.repaymentAmount = repaymentAmount
+			self.renter = renter
+			self.renterName = nil 
+			if renter != nil {
+				self.renterName = FIND.reverseLookup(renter!)
+			}
+			self.tags={}
+			self.scalars={}
+			self.extra={}
+		}
+	}
+
+	pub fun getStorefrontListing(user: Address, id: UInt64) : StorefrontListing? {
+	
+		var listingsV1 : StorefrontListing? = nil
+		let account = getAccount(user)
+		let storefrontCap = account.getCapability<&NFTStorefront.Storefront{NFTStorefront.StorefrontPublic}>(NFTStorefront.StorefrontPublicPath)
+
+		if storefrontCap.check() {
+			let storefrontRef=storefrontCap.borrow()!
+			for listingId in storefrontRef.getListingIDs() {
+				let listing = storefrontRef.borrowListing(listingResourceID: listingId)!
+				let d = listing.getDetails()
+				if d.nftID==id {
+					if !d.purchased {
+						let saleCuts : [StoreFrontCut] = [] 
+						for cut in d.saleCuts {
+							saleCuts.append(
+								StoreFrontCut(
+									amount: cut.amount, 
+									address: cut.receiver.address
+								)
+							)
+						}
+						listingsV1 = StorefrontListing(storefrontID: d.storefrontID, nftType: d.nftType.identifier, nftID: d.nftID, salePaymentVaultType: d.salePaymentVaultType.identifier, salePrice: d.salePrice, saleCuts: saleCuts, customID: nil, commissionAmount: nil, expiry: nil)
+						return listingsV1
+					}
+				}
+			}
+		}
+		return nil
+	}
+
+
+	pub fun getStorefrontV2Listing(user: Address, id: UInt64) : StorefrontListing? {
+		var listingsV2 : StorefrontListing? = nil
+		let account = getAccount(user)
+		let storefrontV2Cap = account.getCapability<&NFTStorefrontV2.Storefront{NFTStorefrontV2.StorefrontPublic}>(NFTStorefrontV2.StorefrontPublicPath)
+
+		if storefrontV2Cap.check() {
+			let storefrontRef=storefrontV2Cap.borrow()!
+			for listingId in storefrontRef.getListingIDs() {
+				let listing = storefrontRef.borrowListing(listingResourceID: listingId)!
+				let d = listing.getDetails()
+				if d.nftID==id {
```

Comment:
i belive this logic is better written in return home early style, it will be flatter and easier to understand IMHO.